### DISCUSSION TITLE
charon[68291]: 12[IKE] RADIUS authentication failed

### DISCUSSION THREAD

**Discussion by chuks-ojinnaka** [strongswan/strongswan#397] [Q&A]
I have ikv2 strongswan setup with a windows radius server. 
It is working fine my windows,mac, and linux clients connects.
Now I want to sign different IP range to different groups e.g staff xxx.xxx.xxx.100-xxx.xxx.xxx.200
all the people in this ad group connects
conn tsg-staff
#        rightsourceip=xxx.16.115.0/24
        rightsourceip=xxx.16.115.200-xxx.16.115.220
        rightgroups2="xxx-xx-tsg-staff"
        auto=add

but all those in this group can't connect
conn cs-students
        #        rightsourceip=xxx.16.115.0/24
        rightsourceip=xxx.16.115.0-xxx.16.115.20
        rightgroups="conistonvpn-students"
        auto=add

 
I am not sure what am doing wrong, could please help me.

the conf files and log files are below:

ipsec.conf:
config setup
        charondebug="ike 2, cfg 2"
#        strictcrlpolicy=yes
        uniqueids = no

conn %default
        mobike=yes
        dpdaction=clear
        closeaction=restart
        dpddelay=2400s
        dpdtimeout=160s
        fragmentation=yes
#        rekey=no
        rekey=yes
        reauth=yes
        keyexchange=ikev2
#        auto=add
        esp=aes256-sha1,3des-sha1-modp1024!
        ike=aes256-sha256-modp1024,aes256-sha256-modp2048
        ikelifetime=3h
        lifetime=3600s
        forceencaps=yes
        left=%any
        leftid=@coniston.xx.xx.xx.xx
        leftsubnet=0.0.0.0/0
        leftauth=pubkey
#        leftcert=server-cert.pem
        leftcert=coniston.crt
        leftsendcert=always
        lefthostaccess=yes
        leftfirewall=yes
        right=%any
        rightid=%any
        rightauth=eap-radius
        rightsendcert=never
        rightdns=xxx.16.6.19,xxx.16.5.31
        eap_identity=%identity

conn tsg-staff
#        rightsourceip=xxx.16.115.0/24
        rightsourceip=xxx.16.115.200-xxx.16.115.220
        rightgroups2="xxx-xx-tsg-staff"
        auto=add

conn cs-students
        #        rightsourceip=xxx.16.115.0/24
        rightsourceip=xxx.16.115.0-xxx.16.115.20
        rightgroups="conistonvpn-students"
        auto=add

Error log:

FC_PAD_N) N(NON_FIRST_FRAG) SA TSi TSr ]
May 11 16:34:28 coniston charon[68291]: 11[IKE] received retransmit of request with ID 1, retransmitting response
May 11 16:34:28 coniston charon[68291]: 11[NET] sending packet: from xxx.16.3.161[4500] to xxx.16.70.156[4500] (1236 bytes)
May 11 16:34:28 coniston charon[68291]: 11[NET] sending packet: from xxx.16.3.161[4500] to xxx.16.70.156[4500] (1236 bytes)
May 11 16:34:28 coniston charon[68291]: 11[NET] sending packet: from xxx.16.3.161[4500] to xxx.16.70.156[4500] (1236 bytes)
May 11 16:34:28 coniston charon[68291]: 11[NET] sending packet: from xxx.16.3.161[4500] to xxx.16.70.156[4500] (260 bytes)
May 11 16:34:28 coniston charon[68291]: 10[NET] received packet: from xxx.16.70.156[4500] to xxx.16.3.161[4500] (96 bytes)
May 11 16:34:28 coniston charon[68291]: 10[ENC] parsed IKE_AUTH request 2 [ EAP/RES/ID ]
May 11 16:34:28 coniston charon[68291]: 10[IKE] received EAP identity 'ucact50'
May 11 16:34:28 coniston charon[68291]: 10[CFG] RADIUS server 'pillar.xx.xx.xx.xx' is candidate: 210
May 11 16:34:28 coniston charon[68291]: 10[CFG] sending RADIUS Access-Request to server 'pillar.xx.xx.xx.xx'
May 11 16:34:28 coniston charon[68291]: 10[CFG] received RADIUS Access-Challenge from server 'pillar.xx.xx.xx.xx'
May 11 16:34:28 coniston charon[68291]: 10[IKE] initiating EAP_MSCHAPV2 method (id 0x01)
May 11 16:34:28 coniston charon[68291]: 10[ENC] generating IKE_AUTH response 2 [ EAP/REQ/MSCHAPV2 ]
May 11 16:34:28 coniston charon[68291]: 10[NET] sending packet: from xxx.16.3.161[4500] to xxx.16.70.156[4500] (112 bytes)
May 11 16:34:28 coniston charon[68291]: 13[NET] received packet: from xxx.16.70.156[4500] to xxx.16.3.161[4500] (144 bytes)
May 11 16:34:28 coniston charon[68291]: 13[ENC] parsed IKE_AUTH request 3 [ EAP/RES/MSCHAPV2 ]
May 11 16:34:28 coniston charon[68291]: 13[CFG] sending RADIUS Access-Request to server 'pillar.xx.xx.xx.xx'
May 11 16:34:28 coniston charon[68291]: 13[CFG] received RADIUS Access-Challenge from server 'pillar.xx.xx.xx.xx'
May 11 16:34:28 coniston charon[68291]: 13[ENC] generating IKE_AUTH response 3 [ EAP/REQ/MSCHAPV2 ]
May 11 16:34:28 coniston charon[68291]: 13[NET] sending packet: from xxx.16.3.161[4500] to xxx.16.70.156[4500] (xxx bytes)
May 11 16:34:28 coniston charon[68291]: 12[NET] received packet: from xxx.16.70.156[4500] to xxx.16.3.161[4500] (80 bytes)
May 11 16:34:28 coniston charon[68291]: 12[ENC] parsed IKE_AUTH request 4 [ EAP/RES/MSCHAPV2 ]
May 11 16:34:28 coniston charon[68291]: 12[CFG] sending RADIUS Access-Request to server 'pillar.xx.xx.xx.xx'
May 11 16:34:28 coniston charon[68291]: 12[CFG] received RADIUS Access-Reject from server 'pillar.xx.xx.xx.xx'
May 11 16:34:28 coniston charon[68291]: 12[IKE] RADIUS authentication of 'ucact50' failed
May 11 16:34:28 coniston charon[68291]: 12[IKE] EAP method EAP_MSCHAPV2 failed for peer xxx.16.70.156
May 11 16:34:28 coniston charon[68291]: 12[ENC] generating IKE_AUTH response 4 [ EAP/FAIL ]
May 11 16:34:28 coniston charon[68291]: 12[NET] sending packet: from xxx.16.3.161[4500] to xxx.16.70.156[4500] (80 bytes)
May 11 16:34:28 coniston charon[68291]: 12[IKE] IKE_SA tsg-staff[3] state change: CONNECTING => DESTROYING

**Comment by Thermi** [comment]
Please fix the formatting of your post.

**Reply by chuks-ojinnaka** [reply]
Hi Thermi,

Could you explain why when I use rightgroups= <windows AD group> in my ipsec.conf file I don't get any connection, however I get connections when I use rightgroups2=<windows AD group>

when I use rightsgroups I get the error below:
May 11 16:34:28 coniston charon[68291]: 12[NET] received packet: from xxx.16.70.156[4500] to xxx.16.3.161[4500] (80 bytes)
May 11 16:34:28 coniston charon[68291]: 12[ENC] parsed IKE_AUTH request 4 [ EAP/RES/MSCHAPV2 ]
May 11 16:34:28 coniston charon[68291]: 12[CFG] sending RADIUS Access-Request to server 'pillar.xx.xx.xx.xx'
May 11 16:34:28 coniston charon[68291]: 12[CFG] received RADIUS Access-Reject from server 'pillar.xx.xx.xx.xx'
May 11 16:34:28 coniston charon[68291]: 12[IKE] RADIUS authentication of 'ucact50' failed
May 11 16:34:28 coniston charon[68291]: 12[IKE] EAP method EAP_MSCHAPV2 failed for peer xxx.16.70.156
May 11 16:34:28 coniston charon[68291]: 12[ENC] generating IKE_AUTH response 4 [ EAP/FAIL ]
May 11 16:34:28 coniston charon[68291]: 12[NET] sending packet: from xxx.16.3.161[4500] to xxx.16.70.156[4500] (80 bytes)
May 11 16:34:28 coniston charon[68291]: 12[IKE] IKE_SA tsg-staff[3] state change: CONNECTING => DESTROYING



ipsec.conf:
config setup
        charondebug="ike 2, cfg 2"
#        strictcrlpolicy=yes
        uniqueids = no

conn %default
        mobike=yes
        dpdaction=clear
        closeaction=restart
        dpddelay=2400s
        dpdtimeout=160s
        fragmentation=yes
#        rekey=no
        rekey=yes
        reauth=yes
        keyexchange=ikev2
#        auto=add
        esp=aes256-sha1,3des-sha1-modp1024!
        ike=aes256-sha256-modp1024,aes256-sha256-modp2048
        ikelifetime=3h
        lifetime=3600s
        forceencaps=yes
        left=%any
        leftid=@coniston.xx.xx.xx.xx
        leftsubnet=0.0.0.0/0
        leftauth=pubkey
#        leftcert=server-cert.pem
        leftcert=coniston.crt
        leftsendcert=always
        lefthostaccess=yes
        leftfirewall=yes
        right=%any
        rightid=%any
        rightauth=eap-radius
        rightsendcert=never
        rightdns=xxx.16.6.19,xxx.16.5.31
        eap_identity=%identity

conn tsg-staff
#        rightsourceip=xxx.16.115.0/24
        rightsourceip=xxx.16.115.200-xxx.16.115.220
        rightgroups2="xxx-xx-tsg-staff"
        auto=add

conn cs-students
        #        rightsourceip=xxx.16.115.0/24
        rightsourceip=xxx.16.115.0-xxx.16.115.20
        rightgroups="conistonvpn-students"
        auto=add

**Reply by tobiasbrunner** [reply]
That error is not related to these options because `rightgroups`  (or `rightgroups2`) is only applied as constraint _after_ the authentication. So check on your RADIUS server what's going wrong here.

**Comment by tobiasbrunner** [comment]
`rightgroups2` is not the right option as you only use a single authentication round, so use `rightgroups`. However, the problem seems to be on the RADIUS server, which rejects the authentication. So check what's going on there.

**Reply by chuks-ojinnaka** [reply]
Hi Tobias,
Thanks for your support, when I use rightgroups I don't get any connection at all, but when I use rightgroups2 I can get connections. I am a bit confused, I would appreciate suggestions.
Chuks
